lib/Parse/Lexer.cpp

```diff
@@ -1884,12 +1884,23 @@ void Lexer::lexStringLiteral(unsigned CustomDelimiterLen) {
         // Successfully scanned the body of the expression literal.
         ++CurPtr;
         continue;
-      } else if ((*CurPtr == '\r' || *CurPtr == '\n') && IsMultilineString) {
-        // The only case we reach here is unterminated single line string in the
-        // interpolation. For better recovery, go on after emitting an error.
-        diagnose(CurPtr, diag::lex_unterminated_string);
-        wasErroneous = true;
-        continue;
+      } else if (*CurPtr == '\r' || *CurPtr == '\n') {
+        if (IsMultilineString) {
+          // The only case we reach here is unterminated single line string in the
+          // interpolation. For better recovery, go on after emitting an error.
+          diagnose(CurPtr, diag::string_interpolation_unclosed);
+          diagnose(CurPtr, diag::lex_unterminated_string);
+          wasErroneous = true;
+          continue;
+        } else {
+          const char *fixItLoc = CurPtr - 1;
+          if (*fixItLoc == QuoteChar) {
+            fixItLoc -= CustomDelimiterLen;
+          }
+          diagnose(fixItLoc, diag::string_interpolation_unclosed);
```

Comment:
(Optional) In the case of a single-line string, we can also emit the unterminated string literal error if  `CurPtr[-1]` is *not* `QuoteChar`.

Reply:
Bad idea, disregard 😅

Reply:
🤨

Reply:
Example for when it's a bad idea: `let s = "\(x"; print(x)`.